packages/jest-haste-map/src/index.js

```diff
@@ -54,7 +54,7 @@ type Options = {
   extensions: Array<string>,
   forceNodeFilesystemAPI?: boolean,
   hasteImplModulePath?: string,
-  ignorePattern: HasteRegExp,
+  ignorePattern?: ?HasteRegExp,
```

Comment:
Both?

Reply:
It was just convenient to avoid handling optional properties in the options passed from `jest-runtime`. `Options` has it as an optional nullable property and `InternalOptions` has it only nullable (not optional). Not sure if this is considered a bad practice though.